eng/UseStage1Compiler.props

```diff
@@ -0,0 +1,28 @@
+<Project>
+
+  <PropertyGroup>
+    <LanguageTargets Condition="'$(MSBuildProjectExtension)' == '.scmproj'">$(MSBuildThisFileDirectory)../src/Feersum.Sdk/targets/Feersum.Sdk.targets</LanguageTargets>
+  </PropertyGroup>
+
+  <!-- Config for our language targets. We don't want to pull in a compiler from
+       NuGet but instead want to dogfood the current build outputs. -->
+  <PropertyGroup>
+    <NoImplicitFeersumCoreReference>true</NoImplicitFeersumCoreReference>
+    <CompilerConfiguration>$(Configuration)</CompilerConfiguration>
+    <CompilerConfiguration Condition=" '$(CompilerConfiguration)' == '' ">Debug</CompilerConfiguration>
+    <FeersumCompilerPath>$(MSBuildThisFileDirectory)..\src\Feersum.Stage1\bin\$(CompilerConfiguration)\net5.0\Feersum.Stage1.dll</FeersumCompilerPath>
+  </PropertyGroup>
+
+  <!-- Refernece our core library so it is available to import, and `dotnet` copies it to the output directory. -->
+  <ItemGroup>
+    <ProjectReference Include="$(MSBuildThisFileDirectory)..\src\Serehfa\Serehfa.csproj" />
+  </ItemGroup>
+
+  <!-- Ensure the compiler is built so we can run it. -->
+  <Target Name="EnsureCompilerBuilt" BeforeTargets="PrepareForBuild" >
+    <MSBuild Projects="$(MSBuildThisFileDirectory)..\src\Feersum.Stage1\Feersum.Stage1.fsproj"
```

Comment:
This MSBuild doesn't quite work the way we would like. By the looks of things we're racing to build the
compiler for both this and the main Feersum exectuable:

https://dev.azure.com/iwillspeak/GitHub/_build/results?buildId=1073&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=8ef82b3b-1feb-5bbd-06f6-b1f7b5467f03

```
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018: The "CreateAppHost" task failed unexpectedly. [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018: System.IO.IOException: The process cannot access the file '/home/vsts/work/1/s/src/Feersum.Stage1/obj/Release/net5.0/apphost' because it is being used by another process. [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at System.IO.FileStream.Init(FileMode mode, FileShare share, String originalPath) [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite) [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at System.IO.File.Copy(String sourceFileName, String destFileName, Boolean overwrite) [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at Microsoft.NET.HostModel.AppHost.BinaryUtils.CopyFile(String sourcePath, String destinationPath) [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at Microsoft.NET.HostModel.AppHost.HostWriter.CreateAppHost(String appHostSourceFilePath, String appHostDestinationFilePath, String appBinaryFilePath, Boolean windowsGraphicalUserInterface, String assemblyToCopyResorcesFrom) [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at Microsoft.NET.Build.Tasks.CreateAppHost.ExecuteCore() [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
/opt/hostedtoolcache/dotnet/sdk/5.0.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(464,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/vsts/work/1/s/src/Feersum.Stage1/Feersum.Stage1.fsproj]
  
```